components/Themes/examples/supplemental/ThemerTypicalUseSupplemental.m

```diff
@@ -0,0 +1,188 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+/* IMPORTANT:
+ This file contains supplemental code used to populate the examples with dummy data and/or
+ instructions. It is not necessary to import this file to use Material Components for iOS.
+ */
+
+#import <Foundation/Foundation.h>
+
+#import "ThemerTypicalUseSupplemental.h"
+
+@implementation ThemerTypicalUseViewController (Supplemental)
+
+- (instancetype)initWithColorScheme:(MDCColorScheme *)colorScheme {
+  self = [super init];
+  if (self) {
+    [self commonMDCFlexibleHeaderViewControllerInit];
+    self.colorScheme = colorScheme;
+  }
+  return self;
+}
+
+- (void)viewWillAppear:(BOOL)animated {
+  [super viewWillAppear:animated];
+  [self.navigationController setNavigationBarHidden:YES animated:animated];
+}
+
+- (void)viewDidLayoutSubviews {
+  [super viewDidLayoutSubviews];
+  self.scrollView.contentSize = CGSizeMake(self.view.bounds.size.width, 600);
+  
+  self.activityIndicator.center = CGPointMake(self.view.frame.size.width / 2, 100);
+  self.alertButton.center = CGPointMake(self.view.frame.size.width / 2 - 100, 200);
+  self.featureButton.center = CGPointMake(self.view.frame.size.width / 2 + 50, 200);
+  self.slider.center = CGPointMake(self.view.frame.size.width / 2, 300);
+  self.progressView.center = CGPointMake(self.view.frame.size.width / 2, 400);
+  self.floatingButton.center = CGPointMake(self.view.frame.size.width / 2, 500);
+}
+
+- (void)commonMDCFlexibleHeaderViewControllerInit {
```

Comment:
We should likely only using the commonMDCComponentInit pattern inside of MDCComponent.  For instance I would expect this method to be a class method in MDCFlexibleHeaderViewController.  We do that specifically to avoid naming collisions.